sweepai/core/sweep_bot.py

```diff
@@ -120,9 +121,10 @@ def safe_decode(
         contents = repo.get_contents(path, *args, **kwargs)
         if contents.encoding == "none":
             blob = repo.get_git_blob(contents.sha)
-            # this might be more correct but chatgpt said the latter is better
-            # return base64.b64decode(bytearray(blob.content, "utf-8")).decode("utf-8")
-            return base64.b64decode(blob.content).decode("utf-8")
+            try:
+                return base64.b64decode(blob.content).decode(chardet.detect(base64.b64decode(blob.content))['encoding'])
+            except UnicodeDecodeError:
+                return base64.b64decode(blob.content)
```

Comment:
The fallback in the except UnicodeDecodeError block returns the raw base64-decoded content without decoding it to a string, which may not be the intended behavior and could lead to further processing issues.